internal/internal_worker.go

```diff
@@ -1397,12 +1395,22 @@ func getActivityFunctionName(r *registry, i interface{}) string {
 	return result
 }
 
-func getWorkflowFunctionName(r *registry, i interface{}) string {
-	result := getFunctionName(i)
-	if alias, ok := r.getWorkflowAlias(result); ok {
-		result = alias
+func getWorkflowFunctionName(r *registry, workflowFunc interface{}) (string, error) {
+	fnName := ""
+	fType := reflect.TypeOf(workflowFunc)
+	switch getKind(fType) {
+	case reflect.String:
+		fnName = reflect.ValueOf(workflowFunc).String()
+	case reflect.Func:
+		fnName = getFunctionName(workflowFunc)
+		if alias, ok := r.getWorkflowAlias(fnName); ok {
+			fnName = alias
+		}
+	default:
+		return "", fmt.Errorf("invalid type 'workflowFunc' parameter provided, it can be either worker function or name of the worker type: %v", workflowFunc)
```

Comment:
What is "name of the worker type"?
I also believe it is possible to [register an instance of WorkflowDefinitionFactory](https://github.com/temporalio/go-sdk/blob/00fb03096c85ee092304125c0054f2d78de377a8/internal/internal_worker.go#L484). Not sure if it applies in the context of this function.

Reply:
I just copied this line from another place. I think it means "name of the type registered in worker". And it doesn't make sense here for me. I rephrased it to `invalid type 'workflowFunc' parameter provided, it can be either worker function or function name: %v`.